src/Features/CSharp/Portable/UseExpressionBodyForLambda/Analysis.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseExpressionBodyForLambda
+{
+    // Code for the DiagnosticAnalyzer ("Analysis") portion of the feature.
+
+    internal partial class UseExpressionBodyForLambdaCodeStyleProvider
```

Comment:
>UseExpressionBodyForLambdaCodeStyleProvider [](start = 27, length = 43)

Here you use short file names for parts of the partial class. In the abstract class, you use ClassName.ShortName.cs. What is the reason to follow different approaches?

Reply:
It was scrolling off the right side of my solution explorer :)  but i'll go to the other form for consistency.

Reply:
ah, here's teh reason.  I use the A.Y.cs form when it's actually a nested class.  In this case there is no nested class.  It's all jsut one class, just with things separated into parts.  So i named the different parts according to what logic is inside.  I could go with A_Analysis A_Fixing A_Refactoring, etc.  but i'm not sure that will help that much here.

Reply:
Have renamed.

Reply:
Nit: why dot in one case and underscores in another? :-)

---
In reply to: [309022804](https://github.com/dotnet/roslyn/pull/37541#discussion_r309022804) [](ancestors = 309022804)

Reply:
As mentioned above.  `.` is for *nested* types.  `_` is for *partial types*.  There is only one type here.  I just split it up into four files so that each file can focus on one responsibility.